### ISSUE TITLE
DB unexpectedly pads data to word-length

### ISSUE THREAD

**Issue by ranma** [vpelletier/dissn8#5]
If I add single-byte data in the code section like this:
```
    DB  9          ; bLength
    DB  2          ; bDescriptorType (CONFIGURATION)
    DB  0x3b, 0x00 ; wTotalLength
    [...]
```

The actually ROM data produced is: 0x09 0x00 0x02 0x00 0x3b 0x00

To get the correct data into ROM, you currently need to stick to strict multiples of 2 values provided:
```
    DB  9, 2      ; bLength ; bDescriptorType (CONFIGURATION)
    DB  0x3b, 0x00 ; wTotalLength
    [...]    
```

Which then yields the expected: 0x09 0x02 0x3b 0x00

**Comment by vpelletier** [comment]
This one is a bit of a grey-area to me: it is as expected to me, but obscure. This is a consequence of the flash memory not being byte-addressable.

For example, the meaning of setting a label before a `DB` without systematic padding becomes unclear:
```
foo: ; if this label exists...
  DB 9
bar: ; ...then either this label cannot exist, or there is a padding byte
  DB 2
```

Likewise, any non-`DB` following an odd-length `DB` would be either invalid, or cause a padding byte:
```
  DB 9
  MOV A, #1
```

I resolved these issues by forcing alignment after each `DB`, as this felt less surprising than sometimes getting a padding byte by adding a label or an instruction. But admittedly I do not have a lot of experience with well-established assemblers, and how they would deal with such situation.

**Comment by ranma** [comment]
I think the interesting question here is what the SONiX-provided assembler will do in this case, since AFAICS yours is intended to be compatible.
A simple solution would be to only re-align when the next instruction is a word-aligned one, so if you have multiple DB in a row no padding would be added, but I haven't looked at it yet.

**Comment by vpelletier** [comment]
> I think the interesting question here is what the SONiX-provided assembler will do in this case, since AFAICS yours is intended to be compatible.

Good point. I tried the things I mentioned, and it behaves as you expected (no padding after an odd-length `DB` statement followed by another `DB` statement, and padding otherwise). This should be now fixed by 4a4230fca2dbd6002bc146123c2e760715a762cb .

**Comment by ranma** [comment]
> This should be now fixed by 4a4230f .

Great, thanks!
